beacon-chain/execution/block_reader.go

```diff
@@ -207,20 +207,17 @@ func (s *Service) retrieveHeaderInfo(ctx context.Context, bNum uint64) (*types.H
 		return nil, err
 	}
 	if !exists {
-		blk, err := s.eth1DataFetcher.HeaderByNumber(ctx, bn)
+		hdr, err := s.HeaderByNumber(ctx, bn)
 		if err != nil {
 			return nil, err
 		}
-		if blk == nil {
+		if hdr == nil {
 			return nil, errors.Errorf("header with the number %d does not exist", bNum)
 		}
-		if err := s.headerCache.AddHeader(blk); err != nil {
-			return nil, err
-		}
-		info, err = types.HeaderToHeaderInfo(blk)
-		if err != nil {
```

Comment:
Oh cool, we were already storing a subset of the block information in the caches.